src/taxonomy/taxonomy-menu/TaxonomyMenu.jsx

```diff
@@ -0,0 +1,103 @@
+// ts-check
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  useToggle,
+  Button,
+  Dropdown,
+  Icon,
+  IconButton,
+} from '@edx/paragon';
+import { MoreVert } from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+
+import ExportModal from '../export-modal';
+import { importTaxonomyTags } from '../import-tags';
+import messages from './messages';
+
+const TaxonomyMenu = ({
+  taxonomy, iconMenu,
+}) => {
+  const intl = useIntl();
+
+  const getTaxonomyMenuItems = () => {
+    const { systemDefined, allowFreeText } = taxonomy;
+    const menuItems = ['import', 'export'];
+    if (systemDefined) {
+      // System defined taxonomies cannot be imported
+      return menuItems.filter((item) => !['import'].includes(item));
+    }
+    if (allowFreeText) {
+      // Free text taxonomies cannot be imported
+      return menuItems.filter((item) => !['import'].includes(item));
+    }
+    return menuItems;
+  };
```

Comment:
Don't we also need to check that the user is allowed to import changes to this taxonomy?

Reply:
Yep absolutely ok to do this in a follow-up task, I've created https://github.com/openedx/modular-learning/issues/160 for this.